src/components/OpenFileDialog/OpenFileDialog.js

```diff
@@ -0,0 +1,444 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    Box,
+    Modal,
+    ModalTitle,
+    ModalContent,
+    DataTable,
+    DataTableHead,
+    DataTableBody,
+    DataTableRow,
+    DataTableCell,
+    DataTableColumnHeader,
+    DataTableToolbar,
+    NoticeBox,
+    CircularLoader,
+    Button,
+} from '@dhis2/ui'
+import isEqual from 'lodash/isEqual'
+import PropTypes from 'prop-types'
+import React, { useEffect, useMemo, useState } from 'react'
+import { FileList } from './FileList'
+import { NameFilter } from './NameFilter'
+import { styles } from './OpenFileDialog.styles'
+import { OwnerFilter } from './OwnerFilter'
+import { PaginationControls } from './PaginationControls'
+import { VisTypeFilter } from './VisTypeFilter'
+
+const getResourceFromType = type => `${type}s`
```

Comment:
This is def short and compact, but note that this will only work as long as all backend resources are pluralised by exactly suffixing an `s`. Can't say if we'll ever add new resources or if the existing ones will ever change, but if we think we will, maybe its better to get the resource by mapping key/value in an array of all resources 🤷

Reply:
like `const getResourceFromType = type => {visualization: 'visualizations', eventChart: 'eventCharts', eventReport: 'eventReports', map: 'maps'}[type]` (or extract the object as a const if preferred 🙂 )

Reply:
```
const types = {visualization: 'visualizations', eventChart: 'eventCharts', eventReport: 'eventReports', map: 'maps'}
const getResourceFromType = type => types[type]
```

then you could also use the same list in the proptypes, like 
```
type: PropTypes.oneOf(Object.keys(types))
        .isRequired,
```
to make sure that we support all types that are passed in. And it makes it a lot easier to maintain the list of types and resources 👍

Reply:
Added constants for the AO types and a map object in `utils.js` with the endpoint names.